pkg/local_object_storage/writecache/storage.go

```diff
@@ -109,15 +108,16 @@ func (c *cache) deleteFromDB(keys [][]byte) error {
 			if err := b.Delete(keys[i]); err != nil {
 				return err
 			}
-			sz += uint64(len(has))
 			storagelog.Write(c.log, storagelog.AddressField(string(keys[i])), storagelog.OpField("db DELETE"))
 		}
 		return nil
 	})
 	if err != nil {
 		return err
 	}
-	c.objCounters.DecDB()
+	for range keys {
```

Comment:
It is not always possible to delete all objects by keys. I propose, taking into account the lightness of calling `c.objCounters.DecDB()`, to do it right in the transaction above after successful `b.Delete(keys[i])`. Or we can use counter and iterate until its value here.

Reply:
If an error occurs, will the deleted object not be recovered?

Reply:
Oh, yes, ur right, I missed that we only change the internal storage of the BoltDB. According to `Update` docs `the entire transaction is rolled back`, so your code is correct.